src/components/CategoryDetails.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { View, Text, TouchableOpacity, Image, StyleSheet } from "react-native";
+import { CategoryDetailsProps } from "../navigation/NavigationTypes";
+import { FetchRandomMealResponse, Meal } from "../types/types";
+import { FlatList } from "react-native-gesture-handler";
+import { primary } from "../styles/styleGuide";
+import { generalStyles } from "../styles/generalStyleSheet";
+import LinearGradient from "react-native-linear-gradient";
+import Loading from "./Loading";
+import mealService from "../services";
+import { paddings } from "../styles/branding";
+
+const CategoryDetails: FunctionComponent<CategoryDetailsProps> = ({
+  route,
+  navigation,
+}) => {
+  const [categoryDetails, onChangeCategoryDetails] =
+    useState<FetchRandomMealResponse>();
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const categoryName = route.params?.categoryName;
+  const gradientColors = ["transparent", primary.shadowColor];
+  useEffect(() => {
+    navigation.setOptions({ headerShown: false });
+    const getCategoryDetails = async () => {
+      try {
+        setLoading(true);
+        const response = await mealService.fetchCategoriesDetailsApi(
+          categoryName
+        );
+        onChangeCategoryDetails(response);
+      } catch (e) {
+        console.error("Error while fetching categories details", e);
+      } finally {
+        setLoading(false);
+      }
+    };
+    getCategoryDetails();
+  }, []);
+
+  const renderItem = ({ item }: { item: Meal }) => (
+    <TouchableOpacity style={styles.renderItemStyle}>
+      <Image style={styles.imageStyle} source={{ uri: item.strMealThumb }} />
+      <Text numberOfLines={2} style={styles.textStyle}>
+        {item.strMeal}
+      </Text>
+    </TouchableOpacity>
+  );
+
+  return (
+    <View style={[generalStyles.mainViewStyle, styles.mainViewStyle]}>
+      <View style={styles.categoriesViewStyle}>
+        {isLoading ? (
+          <View style={styles.loadingStyle}>
+            <Loading />
+          </View>
+        ) : (
+          <FlatList
+            ListHeaderComponent={() => {
+              return (
+                <LinearGradient
+                  colors={gradientColors}
+                  style={styles.gradientStyle}
+                  start={{ x: 1, y: 0 }}
+                  end={{ x: 1, y: 1 }}
+                >
+                  <Text style={styles.titleStyle}>{categoryName}</Text>
+                  <Image
+                    source={require("../../assets/images/background_image.png")}
+                    style={styles.backgroundImageStyle}
+                  />
+
+                  <View style={styles.topViewStyle}></View>
+                </LinearGradient>
+              );
+            }}
+            data={categoryDetails?.meals}
+            renderItem={renderItem}
+            keyExtractor={(item) => item.idMeal}
+            numColumns={3}
```

Comment:
why?

Reply:
The numColumns is 3 because meals of each category are displayed with a 3 columns layout.